js/freewheel.js

```diff
@@ -475,23 +492,39 @@ OO.Ads.manager(function(_, $) {
               amc.showSkipVideoAdButton(false);
             }
           } else {
-            fwContext.setParameter("renderer.video.clickDetection", true, tv.freewheel.SDK.PARAMETER_LEVEL_GLOBAL);
-            slotStartedCallbacks[ad.ad.getCustomId()] = _.bind(function(adId) {
-                amc.sendURLToLoadAndPlayNonLinearAd(ad, ad.id, null);
-              }, this, ad.id);
-            slotEndedCallbacks[ad.ad.getCustomId()] = _.bind(function(adId) {
-                amc.notifyNonlinearAdEnded(adId);
-              }, this, ad.id);
-            delete adStartedCallbacks[ad.ad.getCustomId()];
-            delete adEndedCallbacks[ad.ad.getCustomId()];
+            _registerDisplayForNonlinearAd();
+            fwContext.setParameter(tv.freewheel.SDK.PARAMETER_RENDERER_VIDEO_CLICK_DETECTION, true, tv.freewheel.SDK.PARAMETER_LEVEL_GLOBAL);
+            adStartedCallbacks[ad.ad.getCustomId()] = _.bind(function(details) {
+              //provide width and height values if available. Alice will use these to resize
+              //the skin plugins div when a non linear overlay is on screen
+              if (details) {
+                ad.width = details.width;
+                ad.height = details.height;
+                //the width and height of the fake video must be equal
+                //to or greater than the size of the overlay creative
+                //or else FW won't display it
+                if (fakeVideo) {
+                  fakeVideo.style.width = details.width + 'px';
+                  fakeVideo.style.height = details.height + 'px';
+                }
+              }
+              amc.sendURLToLoadAndPlayNonLinearAd(ad, ad.id, null);
+            }, this);
+
+            adEndedCallbacks[ad.ad.getCustomId()] = _.bind(function(adId) {
+              if (fakeVideo && overlayContainer) {
+                overlayContainer.removeChild(fakeVideo);
+                fakeVideo = null;
+              }
+              amc.notifyNonlinearAdEnded(adId);
+            }, this, ad.id);
+            delete slotStartedCallbacks[ad.ad.getCustomId()];
+            delete slotEndedCallbacks[ad.ad.getCustomId()];
           }
 
           // Register the content video wrapper to align the overlay to the correct elements
           // TODO: We also need to call this on vpaid
           // getCurrentAdInstance()._creativeRenditions[0].getCreativeApi() == "VPAID"
-          if (!ad.isLinear) {
```

Comment:
I think someone forgot to remove this before. The display for the nonlinear ad was registered a few lines above. There was no point to reregister it again down here since nothing has changed for the elements required for the display. I'll remove the comment.